drivers/serial/uart_nrf_sw_lpuart.c

```diff
@@ -12,6 +12,10 @@
 
 LOG_MODULE_REGISTER(lpuart, CONFIG_NRF_SW_LPUART_LOG_LEVEL);
 
+#ifndef CONFIG_NRF_SW_LPUART_INT_DRV_TX_BUF_SIZE
+#define CONFIG_NRF_SW_LPUART_INT_DRV_TX_BUF_SIZE 0
```

Comment:
Zero-length arrays is a gcc extension, so this may make troubles for others compilers.
Instead of such artificial constructs, why don't you just wrap the declaration of `struct lpuart_int_driven` and all stuff that uses it in `#if IS_ENABLED(CONFIG_NRF_SW_LPUART_INT_DRIVEN)`? Instead of `data->int_driven->`, you could use `data->int_driven.` then, which would be a bit more effective.

Reply:
done